app/components/layout.js

```diff
@@ -68,16 +66,22 @@ class Layout extends React.Component {
           </div>
         </div>
 
-        <Button onClick={this.connectToServer} value='Connect To Server' />
+        <Button onClick={this.connectToServer.bind(this)} value={this.props.connectionStatus} />
+
+        <Table />
       </div>
     )
   }
 }
 
-Layout.propTypes = {
-  // Validation logic
-};
+function mapDispatchToProps(dispatch) {
+  return {
+    actions: bindActionCreators(actions, dispatch)
+  }
+}
 
-window.onload = function(){
-  ReactDOM.render(<Layout/>, document.getElementById('layout-container'));
+function mapStateToProps(state) {
+  return state
 }
+
+export default connect(mapStateToProps, mapDispatchToProps)(Layout)
```

Comment:
1. I don't know, actually. Just a feeling

At first, I didn't understand the double parens style. I think I get it now. Correct me if I'm wrong. `connect()` returns a function which gets applied with the `Layout` object?